### DISCUSSION TITLE
关于软件使用的Java版本

### DISCUSSION THREAD

**Discussion by CloudyEagle** [PBH-BTN/PeerBanHelper#553] [讨论]
我的电脑上曾经开发过Java 8有配置过环境，而发现该用NoJava的版本用不了，从而才发现Java版本不适配。
翻阅过去的发版内容，发现在V4的版本开始使用了高版本的Java，请问除了发版所述的内容，还有其他什么原因不用旧版本的Java。

> 破坏性更改
> - PeerBanHelper 现在需要使用 Java 21 或者更高版本

> 性能优化
> - 现在积极使用 Java 21 提供的虚拟线程特性，大幅度改善在较弱性能设备上，并发请求的性能和资源消耗，并将并发优化应用在了更多的位置

**Comment by Ghost-chu** [comment]
我们积极使用高版本 Java 所提供的新技术和新语言特性。当下和未来都没有使用旧版本 Java 的计划。NoJava 目前仅支持 JDK 21，但在未来我们可能会向上扩展支持范围。
